python/cuml/decomposition/pca.pyx

```diff
@@ -736,3 +739,9 @@ class PCA(Base):
             'X_types_gpu': ['2darray', 'sparse'],
             'X_types': ['2darray', 'sparse']
         }
+
+    def _check_is_fitted(self, attr):
+        if not hasattr(self, attr) or (getattr(self, attr) is None):
```

Comment:
Yeah, agreed on the generalization - good suggestion from Dante

Reply:
I will do the generalization of `NotFittedError` in a following PR